### DISCUSSION TITLE
sdbus method call stuck/block

### DISCUSSION THREAD

**Discussion by jayvardhan-jagarwal** [Kistler-Group/sdbus-cpp#424] [Q&A]
Hi,
I have set withTimeout(2000000us) in xml. And this is a sync communication.
Actually i am trying to call one  method over sdbus. but the problem is i need to do 5 retry. after first try i get timeout with below error.
** 14833 2024/03/25 19:18:52.893543 26992.0040 0  3221 log info verbose 2 lumsum : [org.freedesktop.DBus.Error.Timeout] Connection timed out.**
But in second try these below function call got stuck.
 std::tuple<int32_t, int32_t, int32_t> lumsum(const int32_t& _, const int32_t& _, const std::string& _)
    {
        std::tuple<int32_t, int32_t, int32_t> result;
        proxy_->callMethod("lumsum").onInterface(INTERFACE_NAME).withArguments(_, _, _).storeResultsTo(_);
        return result;
    }
this method call got blocked. and i am expecting time out error like above . 
Hi  @sangelovic plese help me..

**Comment by sangelovic** [comment]
Hey @jayvardhan-jagarwal. Still having issues?
